run_all_experiments.py

```diff
@@ -232,6 +233,10 @@ def parse_args() -> argparse.Namespace:
       default=0,
       help=('Delay each experiment by certain seconds (e.g., 10s) to avoid '
             'exceeding quota limit in large scale experiments.'))
+  parser.add_argument('-p',
+                      '--prompt-builder',
+                      help='The prompt builder to use for harness generation.',
+                      default='DEFAULT')
```

Comment:
Could we use the original builder and automatically determine the prompt content based on [the language of the benchmark](https://github.com/google/oss-fuzz-gen/blob/main/benchmark-sets/comparison/avahi.yaml#L24)?
This will be a bit more intrusive, but it's OK for this case because we plan to run a full evaluation experiment before merging the PR. If your method works better, we can easily replace the current one with yours without further testing.

Your new builder has two new features:
1. A new prompt
2. More context

Comments above suggested:
1. We can create a new prompt by a) merging your prompt and the existing one, and b) changing all `C++` to `C`. Once we prove its impact, we can make it flexible (e.g., use `{LANGUAGE}` and `{LANGUAGE-SPECIFIC-FILLER}` in prompt).
2. We can have a more controlled experiment if the context is separated into a different PR.

Once we address these two comments and adjust the prompt based on benchmark language, both prompt builders are effectively the same, i.e., no need to have this command line arg.

Please let me know if I missed anything or if there is a better solution.

Reply:
Re new prompt then I left my thoughts on why I'd prefer to go this route here: https://github.com/google/oss-fuzz-gen/pull/338#discussion_r1642229322

Also added comments on the high-level steps I think should be taken here: https://github.com/google/oss-fuzz-gen/issues/337